web/src/main/java/com/epam/esm/exception/ControllerExceptionHandler.java

```diff
@@ -45,37 +44,16 @@ public ExceptionResponse entityNotFoundHandler(EntityNotFoundException e, WebReq
     @ResponseStatus(HttpStatus.CONFLICT)
     public ExceptionResponse entityExistsHandler(EntityAlreadyExistsException e, WebRequest request) {
         String localeString = request.getHeader(HttpHeaders.ACCEPT_LANGUAGE);
-        Locale locale = Locale.forLanguageTag(Objects.requireNonNull(localeString));
+        Locale locale = localeString != null ? Locale.forLanguageTag(localeString) : Locale.getDefault();
         String message = messageSource.getMessage(ENTITY_ALREADY_EXISTS, new Object[]{}, locale) + e.getMessage();
         return new ExceptionResponse(HttpStatus.CONFLICT.value(), Collections.singletonList(message));
     }
 
-    @ExceptionHandler(DeleteEntityException.class)
-    @ResponseStatus(HttpStatus.BAD_REQUEST)
-    public ExceptionResponse entityDeleteHandler(DeleteEntityException e, WebRequest request) {
-        String localeString = request.getHeader(HttpHeaders.ACCEPT_LANGUAGE);
-        Locale locale = Locale.forLanguageTag(Objects.requireNonNull(localeString));
-        String message = messageSource.getMessage(DELETE_EXCEPTION, new Object[]{}, locale) + e.getMessage();
-        return new ExceptionResponse(HttpStatus.BAD_REQUEST.value(), Collections.singletonList(message));
-    }
-
-    @ExceptionHandler(MethodArgumentNotValidException.class)
-    @ResponseStatus(HttpStatus.BAD_REQUEST)
-    public ExceptionResponse validatorExceptionHandler(MethodArgumentNotValidException e, WebRequest request) {
-        String localeString = request.getHeader(HttpHeaders.ACCEPT_LANGUAGE);
-        Locale locale = Locale.forLanguageTag(Objects.requireNonNull(localeString));
-        BindingResult bindingResult = e.getBindingResult();
-        String errorMessage = Objects.requireNonNull(bindingResult.getFieldError()).getDefaultMessage();
-        String errorField = Objects.requireNonNull(bindingResult.getFieldError()).getField();
-        String message = messageSource.getMessage(VALIDATOR_EXCEPTION, new Object[]{}, locale) + errorField + DELIMITER + errorMessage;
-        return new ExceptionResponse(HttpStatus.BAD_REQUEST.value(), Collections.singletonList(message));
-    }
-
     @ExceptionHandler(BindException.class)
```

Comment:
Why you change the exception type?

Reply:
They both have the same internal state more or less. MethodArgumentNotValidException
extends BindException. This way I can handle more exceptions.